### PR TITLE
Fix various non-critical issues

### PR DESCRIPTION
**Pull request by kerneltoast**

### CODE REVIEW

File: lab4/provided/processor.cc [hunk 1]
```diff
@@ -20,19 +21,15 @@ Processor::~Processor()
...
 Processor::sendRequest(Record &r)
 {
-    uint8_t *data = nullptr;
     if (r.write) {
-        data = new uint8_t[r.size];
-        for (int i=0; i<r.size; i++) {
-            data[i] = (uint8_t)(r.requestId & 0xFF);
-        }
+        r.data = new uint8_t[r.size];
+        memset(r.data, (uint8_t)r.requestId, r.size);
```
Comment (powerjg) [inline lab4/provided/processor.cc:28]: The memset change is good here, but I'd rather not push the other changes. I'm planning on adding some testing code that will use data soon.

File: lab4/provided/Makefile [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+	memory.o \
+	non_blocking.o \
+	processor.o \
+	set_assoc.o \
+	tag_array.o \
+	ticked_object.o
+
+DEPFLAGS = -MMD -MF $(@:.o=.d)
+deps := $(patsubst %.o,%.d,$(objs))
+-include $(deps)
```
Comment (powerjg) [inline lab4/provided/Makefile:24]: This is more makefile magic than I know...

**Review by powerjg — COMMENTED** [review]
Hi Sultan,

Most of the changes look good to me. One minor thing that I would rather not merge, though.

Thanks!!

File: lab4/provided/Makefile (hunk 2 above)
Comment (animalvgamer) [reply lab4/provided/Makefile:24]: This looks okay actually, the lines above just quickly set some dependency file references. Honestly my only bad comment is that the -g and debug flags should be kept separate from the main flags, and adjustable by Makefile argument.

Comment (powerjg) [reply lab4/provided/Makefile:24]: +1, good idea.

File: lab4/provided/processor.cc (hunk 1 above)
Comment (powerjg) [reply lab4/provided/processor.cc:28]: I've made some updates here now. Could you look at merging with the current master and see if this still makes sense. If so, I'm happy to merge it.

Comment (kerneltoast) [reply lab4/provided/processor.cc:28]: The updates you made didn't affect this; it looks good to me.

File: lab4/provided/Makefile (hunk 2 above)
Comment (kerneltoast) [reply lab4/provided/Makefile:24]: Done @animalvgamer.

**Comment by powerjg** [comment]
Thanks for the updates!

**Pull request merged**